txpipe/lssweights.py

```diff
@@ -0,0 +1,1103 @@
+		for ibin in range(self.Ntomo):
+			#get weight per lens object
+			subgroup = binned_output[f"lens/bin_{ibin}/"]
+			ra = subgroup["ra"][:]
+			dec = subgroup["dec"][:]
+			pix = hp.ang2pix(self.pixel_metadata['nside'],ra,dec,lonlat=True,nest=True) #can switch to using txpipe tools for this?
+			obj_weight = 1./Fmap_list[ibin][pix]
+			obj_weight[obj_weight==1./hp.UNSEEN] = 0.0
+
+			subgroup["weight"][...] *= obj_weight
+
+			#also plot a histogram of the weights
+			axs[0][ibin].hist(obj_weight, bins=100)
+			axs[0][ibin].set_xlabel('weight')
+			axs[0][ibin].set_yticks([])
+			axs[0][ibin].set_title(f'lens {ibin}')
+		filepath = output_dir.path_for_file(f"weights_hist.png")
+		fig.tight_layout()
+		fig.savefig(filepath)
+		fig.clear()
+		plt.close()
+
+
+
+		#### save the tomography file
+		tomo_output = self.open_output("lens_tomography_catalog", parallel=True)
+		with self.open_input("lens_tomography_catalog_unweighted") as tomo_input:
+			tomo_output.copy(tomo_input["tomography"],"tomography")
+
+		lens_weight = tomo_output['tomography/lens_weight'][...]
+		for ibin in range(self.Ntomo):
+			binned_subgroup = binned_output[f"lens/bin_{ibin}/"]
+			mask = (tomo_output['tomography/bin'][...] == ibin)
+			lens_weight[mask] *= binned_subgroup["weight"][...]
+		tomo_output['tomography/lens_weight'][...] = lens_weight
+
+		#### close the outputs
+		tomo_output.close()
+		binned_output.close()
+
+
+
+	def summarize(self, output_dir, density_correlation, weighted_density_correlation, fit_output):
+		"""
+		make 1d density plots and other summary statistics and save them
+		
+		Params
+		------
+		output_dir: string
+
+		density_correlation: lsstools.DensityCorrelation 
+		
+		weighted_density_correlation: lsstools.DensityCorrelation 
+		
+		fit_output: dict
+		"""
+		import numpy as np
+		import h5py 
+		import scipy.stats
+
+		ibin = density_correlation.tomobin
+
+		#make 1d density plots
+		for imap in np.unique(density_correlation.map_index):
+			filepath = output_dir.path_for_file(f"sys1D_lens{ibin}_SP{imap}.png")
+			density_correlation.plot1d_singlemap(filepath, imap, extra_density_correlations=[weighted_density_correlation] )
+
+		#save the 1D density trends
+		dens_corr_file_name = output_dir.path_for_file(f"unweighted_density_correlation_lens{ibin}.hdf5")
+		density_correlation.save_to_hdf5(dens_corr_file_name)
+		weighted_dens_corr_file_name = output_dir.path_for_file(f"weighted_density_correlation_lens{ibin}.hdf5")
+		weighted_density_correlation.save_to_hdf5(weighted_dens_corr_file_name)
+
+		#save map names, coefficients and chi2 from the fit into an hdf5 file
+		fit_summary_file_name = output_dir.path_for_file(f"fit_summary_lens{ibin}.hdf5")
+		fit_summary_file = h5py.File(fit_summary_file_name, 'w')
+		fit_summary_file.file.create_group("fit_summary")
+		if len(fit_output['sig_map_index']) != 0:
+			fit_summary_file["fit_summary"].create_dataset('fitted_map_id', data=fit_output['sig_map_index'])
```

Comment:
can tidy this using `summary_group = fit_summary_file["fit_summary"]` then reusing `summary_group`.